ckan/lib/search/index.py

```diff
@@ -50,11 +50,12 @@ def escape_xml_illegal_chars(val, replacement=''):
     return _illegal_xml_chars_re.sub(replacement, val)
 
 
-def clear_index():
+def clear_index(defer_commit=True):
```

Comment:
See comment above regarding `defer_commit` and clear. All the changes in this file can be reversed